README.md

```diff
@@ -53,7 +53,47 @@ cd onyo
 pip install -e ".[tests, docs]"
 ```
 
-### Tests
+### Design
+
+Onyo follows a set of development and coding conventions throughout its code
+base. For linting and type checking, the Onyo project uses flake8 and Pyre. The
+following sections describe additional conventions followed in this project,
+extending the existing [PEP 8 Style Guide](https://peps.python.org/pep-0008/).
+
+#### The Complete Code Base:
+- All classes, functions and properties have a doc-string describing their
+  functionality.
+- All functions have type hinting (including explicit return type for `None`).
+- The code conforms to [PEP 8](https://peps.python.org/pep-0008/).
+
+#### lib Tests (`tests/lib/`):
+- follow Onyo's code conventions, see [The Complete Code Base](#the-complete-code-base).
+- define the expected behavior of the code for all use-cases of the different
+  consumers (i.e. CLI, demos).
+- all public functions and properties of all classes are tested with all
+  possible combinations of input:
+  - single valid inputs, including different data-types (if allowed)
+  - multiple valid inputs (if allowed)
+  - without input (if allowed)
+  - one test for each possible error
+  - one test for each possible conflict with other inputs
+
+#### CLI Tests (`tests/commands/`):
+- follow Onyo's code conventions, see [The Complete Code Base](#the-complete-code-base).
+- define the expected behavior which the user can expect from Onyo.
+- are created completely independent and without any consideration of the code
+  base.
+- cover a large variety of use-cases to verify that Onyo works as intended,
+  even if large parts of the code would be modified.
+- call `subprocess.run(["onyo", ...])` to simulate how Onyo behaves.
```

Comment:
This feels vague here. I think what you mean is that the Onyo API should not be used when testing the CLI, and exclusively onyo CLI commands should be used.

If so, then IMO the inverse of the rule should be added to the API tests.

Reply:
I think in general the usage of the API should not be prohibited during the CLI tests, except for the one specific command tested, i.e. setting some config value with `Repo.config()` in `commands/test_set.py` seems a reasonable thing to do to really just test the functionality of `onyo set` and not entangle it with other commands.

Though you are right, I will add the opposite to the API tests!